array.c

```diff
@@ -1274,20 +1321,63 @@ rb_ary_pop(VALUE ary)
 
 /*
  *  call-seq:
- *     ary.pop    -> obj or nil
- *     ary.pop(n) -> new_ary
+ *    ary.pop -> obj or nil
+ *    ary.pop(n) -> new_array
  *
- *  Removes the last element from +self+ and returns it, or
- *  +nil+ if the array is empty.
+ *  Removes and returns trailing elements from <tt>ary</tt>.
  *
- *  If a number +n+ is given, returns an array of the last +n+ elements
- *  (or less) just like <code>array.slice!(-n, n)</code> does. See also
- *  Array#push for the opposite effect.
+ *  Argument <tt>n</tt>, if given, must be an
+ *  {Integer-convertible object}[doc/implicit_conversion_rdoc.html#label-Integer-Convertible+Objects].
  *
- *     a = [ "a", "b", "c", "d" ]
- *     a.pop     #=> "d"
- *     a.pop(2)  #=> ["b", "c"]
- *     a         #=> ["a"]
+ *  ---
+ *
+ *  When no argument is given,
+ *  removes and returns the last element in <tt>ary</tt>
+ *  unless <tt>ary</tt> is empty:
+ *
+ *    a = [:foo, 'bar', 2]
+ *    a.pop # => 2
+ *    a # => [:foo, "bar"]
+ *
+ *  Returns <tt>nil</tt> if <tt>ary</tt> is empty:
+ *
+ *    a = []
+ *    a.pop # => nil
+ *
+ *  ---
+ *
+ *  When argument <tt>n</tt> is given and <tt>0 <= n <= ary.size</tt>,
```

Comment:
How do you feel about writing this out?

I'm not sure myself: "When argument n is positive and less than or equal to the array size,"

Reply:
Both lack grace.  We need something that's immediately recognizable.  Can we establish the idea of n's being 'in range'?
I don't like any of these, either:
- When n is an index in ary.
- When ary[n] exists.
- When n is in (0..ary.size-1).

Reply:
We already have a couple of references to an index's being in or out of range.

I'd like to add a small section, "Array Indexes," to the very top of the page.  There, I'd define in-range and out-of-range for both positive and negative indexes.  That would allow us to use plain English to talk about indexes.

Makes sense?

Reply:
I agree that my solution lacks grace, I think an "Array Indexes" section would be better as it also provides knowledge about behaviors of many methods, even outside Array

Reply:
Done.